zerver/lib/push_notifications.py

```diff
@@ -396,6 +425,7 @@ def add_push_device_token(user_profile: UserProfile,
             'user_id': user_profile.id,
             'token': token_str,
             'token_kind': kind,
+            'encrypted': encrypt_notifications,
```

Comment:
@hashirsarwar if you want to fix that test failure that you've been facing you'll need to change this line to:
```python3
'encrypted': ujson.encode(encrypted),
```

Take a quick look at `send_analytics_to_remote_server` from `zerver/lib/remote_server.py` for why I'm suggesting this. Also, if you haven't already, also take a look at the [discussion](https://chat.zulip.org/#narrow/stream/3-backend/topic/potential.20send_to_push_bouncer.20bug) I had with Anders and Steve on  CZO).

*Sidenote*: TBH this feels more like a quick fix. I expected `send_json_to_push_bouncer` to properly json encode the data and send it. Instead I get this rather complicated traceback:
```
FAIL: test_push_bouncer_api (zerver.tests.test_push_notifications.PushBouncerNotificationTest)
...
  File "/srv/zulip/zerver/lib/push_notifications.py", line 436, in add_push_device_token
    send_json_to_push_bouncer('POST', 'push/register', post_data)
  File "/srv/zulip/zerver/lib/remote_server.py", line 101, in send_json_to_push_bouncer
    extra_headers={"Content-type": "application/json"},
  File "/srv/zulip/zerver/lib/remote_server.py", line 59, in send_to_push_bouncer
    headers=headers)
  File "/usr/lib/python3.6/unittest/mock.py", line 939, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/usr/lib/python3.6/unittest/mock.py", line 1005, in _mock_call
    result = effect(*args, **kwargs)
  File "/srv/zulip/zerver/tests/test_push_notifications.py", line 116, in bounce_request
    subdomain='')
  File "/srv/zulip/zerver/lib/test_classes.py", line 528, in uuid_post
    return self.client_post(*args, **kwargs)
  File "/srv/zulip/zerver/lib/test_helpers.py", line 327, in wrapper
    result = f(self, url, info, **kwargs)
  File "/srv/zulip/zerver/lib/test_classes.py", line 240, in client_post
    result = django_client.post(url, info, **kwargs)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/test/client.py", line 543, in post
    response = super().post(path, data=data, content_type=content_type, secure=secure, **extra)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/test/client.py", line 354, in post
    post_data = self._encode_data(data, content_type)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/test/client.py", line 313, in _encode_data
    return encode_multipart(BOUNDARY, data)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/django/test/client.py", line 194, in encode_multipart
    for (key, value) in data.items():
AttributeError: 'str' object has no attribute 'items'
```

When I did a bit of print-debugging to check the headers at different parts of the request (during the test) it seems like the `Content-Type` header was being overwritten leading Django to raise this error. In fact, it seems like none of the headers that we set in `send_to_push_bouncer` area actually being sent because when I print (request.headers) under `add_apns_device_token` I don't see the right `User-Agent` nor the right `Content-Type`. So this might be worth investigating. @timabbott do you, off the top of your head, have any idea of what might be happening here?